build/networking/scripts/build_p4c.sh

```diff
@@ -94,7 +94,6 @@ git submodule update --init --recursive
 mkdir build && mkdir -p "${WORKING_DIR}"/p4c/install && cd build
 
 cmake -DCMAKE_INSTALL_PREFIX="${WORKING_DIR}"/p4c/install \
-  -DENABLE_BMV2=OFF \
```

Comment:
Why are we removing this?

Reply:
This was the reason for build failures in container build. Confirmed by the p4c team and needs to be removed

Reply:
> This was the reason for build failures in container build. Confirmed by the p4c team and needs to be removed

What were the build failures?

Reply:
Error Logs:
[ 87%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/midend.cpp.o
In file included from /root/P4C/backends/bmv2/common/helpers.h:22,
                 from /root/P4C/backends/bmv2/common/action.h:20,
                 from /root/P4C/backends/bmv2/psa_switch/psaSwitch.h:20,
                 from /root/P4C/backends/ebpf/psa/ebpfPsaGen.h:20,
                 from /root/P4C/backends/tc/backend.h:20,
                 from /root/P4C/backends/tc/ebpfCodeGen.h:20,
                 from /root/P4C/backends/tc/ebpfCodeGen.cpp:17:
/root/P4C/backends/bmv2/common/expression.h:125:30: error: 'IntMod' in namespace 'IR' does not name a type
  125 |     void postorder(const IR::IntMod *expression) override;
      |                              ^~~~~~
/root/P4C/backends/bmv2/common/expression.h:125:10: error: 'void BMV2::ExpressionConverter::postorder(const int*)' marked 'override', but does not override
  125 |     void postorder(const IR::IntMod *expression) override;
      |          ^~~~~~~~~
In file included from /root/P4C/backends/bmv2/common/helpers.h:22,
                 from /root/P4C/backends/bmv2/common/action.h:20,
                 from /root/P4C/backends/bmv2/psa_switch/psaSwitch.h:20,
                 from /root/P4C/backends/ebpf/psa/ebpfPsaGen.h:20,
                 from /root/P4C/backends/tc/backend.h:20,
                 from /root/P4C/backends/tc/backend.cpp:17:
/root/P4C/backends/bmv2/common/expression.h:125:30: error: 'IntMod' in namespace 'IR' does not name a type
  125 |     void postorder(const IR::IntMod *expression) override;
      |                              ^~~~~~
/root/P4C/backends/bmv2/common/expression.h:125:10: error: 'void BMV2::ExpressionConverter::postorder(const int*)' marked 'override', but does not override
  125 |     void postorder(const IR::IntMod *expression) override;
      |          ^~~~~~~~~
[ 87%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/pnaProgramStructure.cpp.o
[ 87%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/tc.cpp.o
[ 88%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/tcAnnotations.cpp.o
[ 88%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/version.cpp.o
[ 89%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/__/ebpf/ebpfBackend.cpp.o
[ 89%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/__/ebpf/ebpfProgram.cpp.o
[ 89%] Building CXX object backends/tc/CMakeFiles/p4c-pna-p4tc.dir/__/ebpf/ebpfTable.cpp.o
make[2]: *** [backends/tc/CMakeFiles/p4c-pna-p4tc.dir/build.make:95: backends/tc/CMakeFiles/p4c-pna-p4tc.dir/ebpfCodeGen.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [backends/tc/CMakeFiles/p4c-pna-p4tc.dir/build.make:82: backends/tc/CMakeFiles/p4c-pna-p4tc.dir/backend.cpp.o] Error 1
In file included from /root/P4C/backends/bmv2/common/backend.h:22,
                 from /root/P4C/backends/tc/pnaProgramStructure.h:20,
                 from /root/P4C/backends/tc/pnaProgramStructure.cpp:17:
/root/P4C/backends/bmv2/common/expression.h:125:30: error: 'IntMod' in namespace 'IR' does not name a type
  125 |     void postorder(const IR::IntMod *expression) override;
      |                              ^~~~~~
/root/P4C/backends/bmv2/common/expression.h:125:10: error: 'void BMV2::ExpressionConverter::postorder(const int*)' marked 'override', but does not override
  125 |     void postorder(const IR::IntMod *expression) override;
      |          ^~~~~~~~~
In file included from /root/P4C/backends/bmv2/common/helpers.h:22,
                 from /root/P4C/backends/bmv2/common/action.h:20,
                 from /root/P4C/backends/bmv2/psa_switch/psaSwitch.h:20,
                 from /root/P4C/backends/ebpf/psa/ebpfPsaGen.h:20,
                 from /root/P4C/backends/tc/backend.h:20,
                 from /root/P4C/backends/tc/tc.cpp:17:
/root/P4C/backends/bmv2/common/expression.h:125:30: error: 'IntMod' in namespace 'IR' does not name a type
  125 |     void postorder(const IR::IntMod *expression) override;
      |                              ^~~~~~
/root/P4C/backends/bmv2/common/expression.h:125:10: error: 'void BMV2::ExpressionConverter::postorder(const int*)' marked 'override', but does not override
  125 |     void postorder(const IR::IntMod *expression) override;
      |          ^~~~~~~~~
In file included from /root/P4C/backends/bmv2/common/helpers.h:22,
                 from /root/P4C/backends/bmv2/common/action.h:20,
                 from /root/P4C/backends/bmv2/psa_switch/psaSwitch.h:20,
                 from /root/P4C/backends/ebpf/ebpfBackend.cpp:19:
/root/P4C/backends/bmv2/common/expression.h:125:30: error: 'IntMod' in namespace 'IR' does not name a type
  125 |     void postorder(const IR::IntMod *expression) override;
      |                              ^~~~~~
/root/P4C/backends/bmv2/common/expression.h:125:10: error: 'void BMV2::ExpressionConverter::postorder(const int*)' marked 'override', but does not override
  125 |     void postorder(const IR::IntMod *expression) override;
      |          ^~~~~~~~~
make[2]: *** [backends/tc/CMakeFiles/p4c-pna-p4tc.dir/build.make:134: backends/tc/CMakeFiles/p4c-pna-p4tc.dir/pnaProgramStructure.cpp.o] Error 1
[ 89%] Linking CXX executable p4c-dpdk
make[2]: *** [backends/tc/CMakeFiles/p4c-pna-p4tc.dir/build.make:186: backends/tc/CMakeFiles/p4c-pna-p4tc.dir/__/ebpf/ebpfBackend.cpp.o] Error 1
[ 89%] Built target p4c-dpdk
make[2]: *** [backends/tc/CMakeFiles/p4c-pna-p4tc.dir/build.make:147: backends/tc/CMakeFiles/p4c-pna-p4tc.dir/tc.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1537: backends/tc/CMakeFiles/p4c-pna-p4tc.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
The command '/bin/sh -c /root/scripts/distro_pkg_install.sh --install-dev-pkgs --scripts-dir=/root/scripts &&     /root/scripts/install_nr_modules.sh --workdir=/root &&     /root/scripts/post_build_cleanup.sh --workdir=/root --keep-source-code="${KEEP_SOURCE_CODE}"        --deployment-image="${DEPLOYMENT_IMAGE}"' returned a non-zero code: 1


has context menu

Reply:
Address two build issues:
- Missing 'netaddr' module. Updated install script to provide it.
- Compile errors in P4C. The latest P4C compiler uses some code from bmv2 , so it shouldn't be disabled. Removed ENABLE_BMV2=OFF from the cmake command. Verified with compiler team that this is correct.

Reply:
The critical one here is the change to the ENABLE_BMV2 parameter. The reason for adding the 'netaddr' module can reasonably be induced. The ENABLE_BMV2 is considerably more subtle. The most important reason for adding a comment is to answer the question, "Why?"